tests/test_api_flows.py

```
@@ -0,0 +1,629 @@
+        # For this test, we assume manual closure of an "open" draw is fine.
+        # If a test needs to wait for scheduled_close_time, mocking time (e.g. with freezegun) is needed.
+
+        close_response = client.post(f"/api/draws/close/{draw_id}")
+        assert close_response.status_code == 200, close_response.text
+        closed_draw_data = close_response.json()
+        assert closed_draw_data["status"] == "completed"
+        assert closed_draw_data.get("ledger_hash") is not None
+        assert "winners_by_tier" in closed_draw_data
+
+        if closed_draw_data["participants"]:
+             assert len(closed_draw_data["winners_by_tier"]) == 1
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)